### DISCUSSION TITLE
Security concern? Public supabase client?

### DISCUSSION THREAD

**Discussion by Vandivier** [supabase/supabase#1760] [Questions] [answered]
AFAIK, I need to initialize the supabase client and pass it to the UI component in order to use social sign on buttons.

Is there a way that I can use the GitHub social sign on button without exposing my supabase secrets to the UI + window object?

![image](https://user-images.githubusercontent.com/5559355/120083707-99a19e80-c098-11eb-9be9-910e213318f7.png)

**Comment by inian** [comment]
What about not creating the variable in global scope and creating it inside an immediately executing function @Vandivier? Something like this
```
(() => {
   const supabase = createClient()
   supabase.auth.signIn()
}())

// supabase will be inaccessible
```

**Reply by Vandivier** [reply]
Hey inian! thanks for the answer. this might work but I'm not sure how the client would be able to get the secrets out of the node process in that case

I listened to the deep dive below from Anthony Wilson (u prob know - supa CTO) and he says the anon 'secret' is fine to disclose publically as long as you lock down tables (ig you know this) it's basically a guest secret so nbd.

ref: https://www.youtube.com/watch?v=0LvCOlELs5U

however there is obv the risk if you don't lock tables properly so maybe just don't put this secret in the client

I'm personally handling this in my project by having UI -> cloud function -> supabase -> cloud function -> UI. very excited for supabase workflows that would reduce latency in this regard.

ref: https://github.com/Vandivier/futurecaster/blob/main/framework/futurecaster/api/user.ts

In theory there are a bunch of secure cases where my web client can use supabase client directly, and others where it can't. In practice this would mean mixing and matching UI clients which is no fun, so I just always call my own server and my own server can call a third party where applicable

**Reply by inian** [reply]
yeah as long as your tables are protected by RLS and realtime is disabled on sensitive tables, there is no issue with exposing your anon key!

**Comment by Vandivier** [comment, marked as answer]
I think this is the answer currently: https://github.com/supabase/supabase/discussions/1760#discussioncomment-803838

**Reply by Vandivier** [reply]
I do want to add that it seems like the jwt session access token contains the user's unencrypted email atm...not sure that's secure either to me, but some people think it's nbd ig: https://stackoverflow.com/a/42387439/3931488
